### DISCUSSION TITLE
assert.js, notDeepStrictEqual (line 547) contains confusing language

### DISCUSSION THREAD

**Discussion by subtly-sms** [nodejs/help#4714] [Q&A] [answered]
The notDeepStrictEqual() function contained within assert.js (https://github.com/nodejs/node/blob/v19.9.0/lib/assert.js#L547), has a test that checks for equality between an actual and expected value using the isDeepStrictEqual() function on line 547. However, the error returned references 'notDeepStrictEqual' as the operator and stackStartFn.

The language of the innerFail seems misleading because the failure states 'notDeepStrictEqual' but in the case of failure the actual and expected values are indeed equal. The logic of the code makes sense -> when testing for inequality, the numbers should not be equal. But the messaging appears to say the opposite -> actual and expected values match, but the innerFail error could imply that the values should not match, literally 'notDeepStrictEqual'.

**Comment by subtly-sms** [comment]
Rephrasing this as a question for discussion, does the solid logic behind the code invalidate any potential concerns about messaging?

**Reply by Trott** [reply, marked as answer]
> Rephrasing this as a question for discussion, does the solid logic behind the code invalidate any potential concerns about messaging?

Answering your question with another question: Is there an actual situation where the message provided to the user is in fact misleading? This sure seems to make a lot of sense to me:

```console
> assert.notDeepStrictEqual(1, 1)
Uncaught:
AssertionError [ERR_ASSERTION]: Expected "actual" not to be strictly deep-equal to: 1
```

**Reply by subtly-sms** [reply]
Looking at the code, it looks like you can add your own custom message, which means you won't necessarily receive that specific assertion error in all cases, right?
